zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -336,16 +421,75 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             call.rw_counter_end_of_reversion,
             call.is_persistent,
         )?;
+
         self.sufficient_gas_left
             .assign(region, offset, F::from(tx.gas - step.gas_cost))?;
-        self.transfer_with_gas_fee.assign(
+
+        let add_tx_value_and_mul_gas_fee_by_gas = tx.value + gas_fee;
+        self.add_tx_value_and_mul_gas_fee_by_gas_gadget.assign(
             region,
             offset,
-            caller_balance_pair,
-            callee_balance_pair,
-            tx.value,
-            gas_fee,
+            [tx.value, gas_fee],
+            add_tx_value_and_mul_gas_fee_by_gas,
         )?;
+
+        self.balance_not_enough.assign(
+            region,
+            offset,
+            caller_balance_pair.1.clone(),
+            add_tx_value_and_mul_gas_fee_by_gas,
+        )?;
+
+        if tx.invalid_tx >= 1 {
```

Comment:
A lot of code duplicated in this if, so would remove this `if` and either just use inline code like
```
self.intrinsic_tx_value
    .assign(region, offset, Some(if tx.invalid_tx >= 1 { U256::zero() } else { tx.value }.to_le_bytes()))?;
```

or define some extra variables that are different in both cases like this so you can reuse all the main code:
```
let intrinsic_tx_value = if tx.invalid_tx >= 1 { U256::zero() } else { tx.value };
...
self.intrinsic_tx_value
  .assign(region, offset, Some(intrinsic_tx_value.to_le_bytes()))?;

```

Reply:
Update. Remove duplicated code.